source/extensions/transport_sockets/tls/ssl_socket.cc

```diff
@@ -377,6 +377,31 @@ absl::optional<SystemTime> SslSocket::expirationPeerCertificate() const {
   return Utility::getExpirationTime(*cert);
 }
 
+const std::string& SslSocket::urlEncodedPemEncodedPeerCertificateChain() const {
+  if (!cached_pem_encoded_peer_cert_chain_.empty()) {
+    return cached_pem_encoded_peer_cert_chain_;
+  }
+
+  STACK_OF(X509)* cert_chain = SSL_get_peer_cert_chain(ssl_.get());
+  for (uint64_t i = 0; i < sk_X509_num(cert_chain); i++) {
+    X509* cert = sk_X509_value(cert_chain, i);
+
+    bssl::UniquePtr<BIO> buf(BIO_new(BIO_s_mem()));
+    RELEASE_ASSERT(buf != nullptr, "");
+    RELEASE_ASSERT(PEM_write_bio_X509(buf.get(), cert) == 1, "");
+    const uint8_t* output;
+    size_t length;
+    RELEASE_ASSERT(BIO_mem_contents(buf.get(), &output, &length) == 1, "");
+
+    absl::string_view pem(reinterpret_cast<const char*>(output), length);
+    cached_pem_encoded_peer_cert_chain_ = absl::StrCat(
+        cached_pem_encoded_peer_cert_chain_,
+        absl::StrReplaceAll(
+            pem, {{"\n", "%0A"}, {" ", "%20"}, {"+", "%2B"}, {"/", "%2F"}, {"=", "%3D"}}));
```

Comment:
This should be simply:
```
cached_pem_encoded_peer_cert_chain_ = absl::StrReplaceAll(
    pem, {{"\n", "%0A"}, {" ", "%20"}, {"+", "%2B"}, {"/", "%2F"}, {"=", "%3D"}}));
```

Reply:
What if there are multiple certs?

Reply:
Oops, you're totally right! This is good and my suggestion was wrong.